test/utils/formatting/dataItem.js

```diff
@@ -0,0 +1,255 @@
+import expect from 'expect.js';
+
+import { formatValueFromStateAndProps } from '../../../src/utils/formatting/dataItem';
+
+const createState = ({
+  symbol = 'MSFT',
+  value,
+  optionalValue,
+  valueSuffix,
+  optionalValueSuffix
+}) => ({
+  selectedStock: symbol,
+  stocks: {
+    MSFT: {
+      stockData: {
+        Div: {
+          value,
+          optionalValue,
+          valueSuffix,
+          optionalValueSuffix
+        }
+      }
+    }
+  }
+});
+
+const createProps = ({
+  label,
+  valuePrecision,
+  optionalValuePrecision
+}) => ({
+  label,
+  valuePrecision,
+  optionalValuePrecision
+});
+
+export const formatValueFromStateAndPropsTest = () => {
+  const VALUE = 3.54;
+  const OPTIONAL_VALUE = 0.45;
+  const VALUE_SUFFIX = '%';
+  const OPTIONAL_VALUE_SUFFIX = '%';
+
+  const VALUE_PRECISION = 2;
+  const OPTIONAL_VALUE_PRECISION = 2;
+
+  describe('formatValueFromStateAndProps', () => {
+    it('should properly format when ALL fields are defined', () => {
+      const state = createState({
+        value: VALUE,
+        optionalValue: OPTIONAL_VALUE,
+        valueSuffix: VALUE_SUFFIX,
+        optionalValueSuffix: OPTIONAL_VALUE_SUFFIX
+      });
+      const ownProps = createProps({
+        label: 'Div',
+        valuePrecision: VALUE_PRECISION,
+        optionalValuePrecision: OPTIONAL_VALUE_PRECISION
+      });
+      expect(formatValueFromStateAndProps(state, ownProps)).to
+        .equal(VALUE + VALUE_SUFFIX + ' (' + OPTIONAL_VALUE + OPTIONAL_VALUE_SUFFIX + ')');
+    });
+    it('should properly format when value is not defined', () => {
+      const state = createState({
+        optionalValue: OPTIONAL_VALUE,
+        valueSuffix: VALUE_SUFFIX,
+        optionalValueSuffix: OPTIONAL_VALUE_SUFFIX
+      });
+      const ownProps = createProps({
+        label: 'Div',
+        valuePrecision: VALUE_PRECISION,
+        optionalValuePrecision: OPTIONAL_VALUE_PRECISION
+      });
+      expect(formatValueFromStateAndProps(state, ownProps)).to
+        .equal('-- (' + OPTIONAL_VALUE + OPTIONAL_VALUE_SUFFIX + ')');
```

Comment:
I think we should have a constant defined somewhere for the whole project: `BLANK_FIELD = '--'`.  Then we can import it in utils/formatting/dataItem.js as well as here, and if we ever want to change the character(s) that represent a blank data field. it needs to only be done in a single place.

Maybe define the constant in src/contants/index.js?  Then later if we have a bazillion contacts we can make other files in the constants directory and import them in index, from which we export them all.

Taking this even further, maybe our labels and precisions should all be constants as well.